server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/FilteringManagement.java

```diff
@@ -29,14 +30,14 @@
 
 public interface FilteringManagement {
 
-    Publisher<Void> defineRulesForUser(Username username, List<Rule> rules);
+    Publisher<Optional<Version>> defineRulesForUser(Username username, List<Rule> rules, Optional<Version> ifInState);
```

Comment:
Why not simply `Publisher<Version>` as a return types ? 

When don't we want to have a version?

Idem below...